servers/mlflowserver/mlflowserver/MLFlowServer.py

```diff
@@ -33,8 +34,11 @@ def predict(
 
         if not self.ready:
             raise requests.HTTPError("Model not loaded yet")
-
-        result = self._model.predict(X)
+        if feature_names is not None and len(feature_names) > 0:
+            df = pd.DataFrame(data=X, columns=feature_names)
+        else:
+            df = pd.DataFrame(data=X)
+        result = self._model.predict(df)
```

Comment:
It seems this may be a breaking change if a model was created with the previous model server. Given this is the case, we probably want to make sure users can specify to run it with numpy array as an option. This could be done in a similar way to how the sklearn model server deals with `predict` vs `predict_proba` functions.

Reply:
gotcha. i'll take a look 👍

Reply:
@meoril Is this taken care of? I'm facing this issue.

```
Traceback (most recent call last):
  File "/opt/conda/envs/mlflow/lib/python3.7/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/opt/conda/envs/mlflow/lib/python3.7/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/opt/conda/envs/mlflow/lib/python3.7/site-packages/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/opt/conda/envs/mlflow/lib/python3.7/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/opt/conda/envs/mlflow/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/opt/conda/envs/mlflow/lib/python3.7/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/opt/conda/envs/mlflow/lib/python3.7/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/opt/conda/envs/mlflow/lib/python3.7/site-packages/seldon_core/wrapper.py", line 54, in Predict
    user_model, requestJson, seldon_metrics
  File "/opt/conda/envs/mlflow/lib/python3.7/site-packages/seldon_core/seldon_methods.py", line 108, in predict
    user_model, features, class_names, meta=meta
  File "/opt/conda/envs/mlflow/lib/python3.7/site-packages/seldon_core/user_model.py", line 209, in client_predict
    return user_model.predict(features, feature_names, **kwargs)
  File "/microservice/MLFlowServer.py", line 37, in predict
    result = self._model.predict(X)
  File "/opt/conda/envs/mlflow/lib/python3.7/site-packages/mlflow/keras.py", line 331, in predict
    predicted.index = dataframe.index
AttributeError: 'numpy.ndarray' object has no attribute 'index'
```